### PR TITLE
Introducing Revapi into build system

### PR DESCRIPTION
**Pull request by JonathanGiles**
Currently it is configured to not fail the build (because it is comparing the 1.0.0-preview.N releases rather than 1.0.0). Once this versioning issue is fixed, we should flip this to fail the build.

### CODE REVIEW

**Comment by weshaggard** [comment]
@mitchdenny have changes been made to the triggering of the internal pipelines? I thought we were overriding those so we could always trigger them without path filters getting in the way?

**Comment by weshaggard** [comment]
@JonathanGiles can you give a little information on this tool and how it works? Does it pull down the latest version of a package and compare the API's with the current version? Does it only do a diff or does it detect breaking changes?

**Comment by JonathanGiles** [comment]
The way Revapi is currently configured to work is to pull down the latest Maven-released artifact, and to compare that against the locally-built artifact. It doesn't diff, it does a deeper analysis where it identifies source and binary breaking changes, as well as potentially breaking and non-breaking changes. We can configure what checks are run, what results in a build failure, and we can whitelist allowed breakages. At present the build is configured to not fail due to the versioning issue we currently have, but once we are past that we should apply the build failure config so builds will fail early when changes are introduced.

**Comment by weshaggard** [comment]
>The way Revapi is currently configured to work is to pull down the latest Maven-released artifact

Is there a way to configure it to only pull GA releases or is that what it already does?

**Comment by JonathanGiles** [comment]
It compares against the latest non-snapshot release. You can specify a format to work around that: https://revapi.org/modules/revapi-maven-plugin/#leaving_out_pre_releases

File: pom.client.xml [hunk 1]
```diff
@@ -645,6 +675,32 @@
           </reportSet>
         </reportSets>
       </plugin>
+
+      <plugin>
+        <groupId>org.revapi</groupId>
+        <artifactId>revapi-maven-plugin</artifactId>
+        <version>0.11.2</version> <!-- {x-version-update;org.revapi:revapi-maven-plugin;external_dependency} -->
```
Comment (JimSuplizio) [inline pom.client.xml]: @conniey is the revapi-java plugin dependency not needed here?

Comment (conniey) [reply pom.client.xml]: No. I checked. It's not needed.

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/AccessTier.java [hunk 2]
```diff
@@ -55,7 +55,7 @@
     /**
      * Static value P60 for AccessTier.
      */
-    public static final AccessTier P60 = fromString("P60");
+    public static final AccessTier P65 = fromString("P60");
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/AccessTier.java]: This is just a validation that revapi work correctly right?

Comment (conniey) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/AccessTier.java]: Yup. It's With the "(REMOVE)" commit. Lol. Trying to see if it'll fail the storage build.

File: eng/revapi.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        "configuration": {
+            "filter": {
+                "packages": {
+                    "regex": true,
+                    "exclude" : ["com\\.azure\\..+\\.implementation(\\..+)?"]
+                }
+            }
+        }
+    },
+    {
```
Comment (hemanttanwar) [inline eng/revapi.json]: We might use reclassify extension in future.   More information here https://revapi.org/modules/revapi-java/differences.html

Comment (conniey) [reply eng/revapi.json]: This file can be updated in the future. I didn't add this because we don't have any breaking changes.

**Pull request merged**